test/spec/app/components/footer/footer.controller.spec.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+
+describe("controller test", function () {
+
+    beforeEach(module("FoodVotingApp"));
+
+    describe("footer controller", function () {
+
+        var $scope;
+
+        beforeEach(inject(function (_$rootScope_, $controller) {
+            $scope = _$rootScope_.$new();
+
+            $controller("FooterController as footerCtrl", {
+                $scope: $scope
+            });
+        }));
+
+        it("should test getAppName function", function () {
+
+            spyOn($scope.footerCtrl, "getAppName").and.returnValue("Food Voting App");
```

Comment:
You should not do this while testing controller. Instead inject `APP_CONSTANT` while creating controller and do something like this:
```
it("should test getAppName function", 
    var appName = $scope.footerCtrl.getAppName();
    expect(appName ).toEqual("Food Voting App");
});
```